webrtc/README.md

```diff
@@ -0,0 +1,355 @@
+  certificate fingerprint (e.g.
+  `/ip6/2001:db8::/udp/1234/webrtc/certhash/<hash>/p2p/<peer-id>`)? _Valid_ in
+  the sense that this generated SDP packet can then be used to establish a
+  WebRTC connection to the remote.
+
+  Yes.
+
+### Browser to Browser
+
+Scenario: Browser _A_ wants to connect to Browser node _B_ with the help of
+server node _R_.
+
+- Replace STUN with libp2p's identify and AutoNAT
+  - https://github.com/libp2p/specs/tree/master/identify
+  - https://github.com/libp2p/specs/blob/master/autonat/README.md
+- Replace TURN with libp2p's Circuit Relay v2
+  - https://github.com/libp2p/specs/blob/master/relay/circuit-v2.md
+- Use DCUtR over Circuit Relay v2 to transmit SDP information
+  1. Transform ICE candidates in SDP to multiaddresses.
+  2. Transmit the set of multiaddresses to the remote via DCUtR.
+  3. Transform the set of multiaddresses back to the remotes SDP.
+  4. https://github.com/libp2p/specs/blob/master/relay/DCUtR.md
+
+#### Open Questions
+
+- Can a browser know upfront its UDP port which it is listening for incoming
+  connections on? Does the browser reuse the UDP port across many WebRTC
+  connections? If that is the case one could connect to any public node, with
+  the remote telling the local node what port it is perceived on.
+
+  No, a browser uses a new UDP port for each `RTCPeerConnection`.
+
+- Can _Browser_ control the lifecycle of its local TLS certificate, i.e. can
+  _Browser_ use the same TLS certificate for multiple WebRTC connections?
+
+  Yes. For the lifetime of the page, one can generate a certificate once and
+  reuse it across connections. See also
+  https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/RTCPeerConnection#using_certificates
+
+- Can two _Browsers_ exchange their SDP packets via a third server node using
+  Circuit Relay v2 and DCUtR? Instead of exchanging the original SDP packets,
+  could they exchange their multiaddr and construct the remote's SDP packet
+  based on it?
+
+## Connection Security
+
+While WebRTC offers confidentiality and integrity via TLS, one still needs to
+authenticate the remote peer by its libp2p identity.
+
+After [Connection Establishment](#connection-establishment):
+
+1. _A_ opens a WebRTC datachannel.
+
+2. _A_ starts a Noise `XX` handshake using _A_'s and _B_'s libp2p identity. See
+   [noise-libp2p](https://github.com/libp2p/specs/tree/master/noise).
+
+   Instead of exchanging the TLS certificate fingerprints on the established
+   Noise channel once the Noise handshake succeeded, _A_ and _B_ use the [Noise
+   Prologue](https://noiseprotocol.org/noise.html#prologue) mechanism, thus
+   saving one round trip.
+
+   More specifically _A_ and _B_ set the Noise _Prologue_ to
+   `libp2p-webrtc-noise:<FINGERPRINTS>` before starting the actual Noise
+   handshake. `<FINGERPRINTS>` is the concatenation of the of the two TLS
+   fingerprints of _A_ and _B_ in their multihash byte representation, sorted in
+   ascending order.
```

Comment:
Implementation note: Testing with the browser, it seems like Chromium only supports ECDSA P-256 or RSA key signatures in the webrtc TLS certificate. We tried using P-228 and P-384 which caused the DTLS handshake to fail with `Illegal Parameter`. Refer: https://github.com/libp2p/go-libp2p/pull/1655/files#diff-95f3f7953b6b10bfc25508238854e4e3119283beebe3789d7a5268827d111454R55

Reply:
I have not included this in the specification. In case you would like to have this included, please comment with a paragraph suggestion.